bloomberg/chef-bcpc
# Remove Apache frontend for RADOSgw

Since switching to Civetweb for RADOSgw, the Apache frontend has gone unused. This PR removes the Apache configuration entirely and makes a few other related changes to improve using RADOSgw, as part of an ongoing quest to simplify things slightly by removing unused features and code paths.

To test RADOSgw locally:

* Install **s3cmd** on your host.
* Set up `/etc/hosts` with two entries, one pointing `s3.bcpc.example.com` to 192.168.100.5, and one pointing `bucketname.s3.bcpc.example.com` also to 192.168.100.5, where **bucketname** is the name of the test bucket you will be creating:
```
192.168.100.5 s3.bcpc.example.com
192.168.100.5 erhudy.s3.bcpc.example.com
```
* Set up an s3cmd configuration file (default location is `$HOME/.s3cfg`). To get an access key, run `keystone ec2-credentials-list`, or `keystone ec2-credentials-create` if no credentials are listed here.
```
[default]
access_key = xxx
check_ssl_certificate = False
check_ssl_hostname = False
host_base = s3.bcpc.example.com
host_bucket = %(bucket)s.s3.bcpc.example.com
secret_key = yyy
use_https = True
```
* Verify `s3cmd ls` and `s3cmd la` do not return a 403 or other error (blank is fine).
* Create your testbucket with `s3cmd mb s3://bucketname`, using the name of the bucket you put into your hosts file.
* Put files in with `s3cmd put blah s3://bucketname`, and take them out with `s3cmd get s3://bucketname/blah`.
